Code/Editor/MainWindow.cpp

```diff
@@ -2137,6 +2160,291 @@ bool MainWindow::focusNextPrevChild(bool next)
     return QMainWindow::focusNextPrevChild(next);
 }
 
+void MainWindow::InitializeActionContext()
+{
+    ActionContextProperties contextProperties;
+    contextProperties.m_name = "O3DE Editor";
+
+    m_actionManagerInterface->RegisterActionContext("", EditorMainWindowActionContextIdentifier, contextProperties, this);
+}
+
+void MainWindow::InitializeActions()
+{
+    auto cryEdit = CCryEditApp::instance();
+
+    // --- Level Actions
+
+    // New Level
+    {
+        ActionProperties actionProperties;
+        actionProperties.m_name = "New Level";
+        actionProperties.m_description = "Create a new level";
+        actionProperties.m_category = "Level";
+
+        m_actionManagerInterface->RegisterAction(
+            EditorMainWindowActionContextIdentifier, "o3de.action.level.new", actionProperties,
+            [cryEdit]()
+            {
+                cryEdit->OnCreateLevel();
+            }
+        );
+    }
+
+    // Open Level
+    {
+        ActionProperties actionProperties;
+        actionProperties.m_name = "Open Level...";
+        actionProperties.m_description = "Open an existing level";
+        actionProperties.m_category = "Level";
+
+        m_actionManagerInterface->RegisterAction(
+            EditorMainWindowActionContextIdentifier, "o3de.action.level.open", actionProperties,
+            [cryEdit]()
+            {
+                cryEdit->OnOpenLevel();
```

Comment:
question: This is way outside the scope of this PR, but do we have an idea about removing the reference to the 'god' cryEdit logic? I'm just wondering could we soon (maybe in a follow-up PR) move the code in `OnOpenLevel` to just be in the lambda? And then we can slowly move towards shrinking `CryEditApp` with the hope of ultimately removing it. These changes are fantastic and move us more in the direction of being able to do this. I don't want this to distract from the overall ActionManager effort but it might be something good to keep in mind and maybe return to before too long 🙂

Reply:
Yes! My intention is actually to move the level-related functionality out of CryEdit entirely and into the Prefab system, including the piping to create/open/save. That should make it easier to move the concept of Level into a Gem down the line, if that's what we want to do. For Slices, we may keep a small handler that still follows the current code flow, but is encapsulated enough to make it super simple to remove when we get there.

I don't really consider this as part of the Action Manager effort, but I will create follow-up tasks to ensure we at least bring up the possibility when this is complete.

Reply:
> My intention is actually to move the level-related functionality out of CryEdit entirely and into the Prefab system, including the piping to create/open/save

That's great to hear! 😄 Thanks for the info 🙂